typescript-selenium/pageobjects/dashboard/NewWorkspace.ts

```diff
@@ -0,0 +1,109 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { injectable, inject } from "inversify";
+import { DriverHelper } from "../../utils/DriverHelper";
+import { CLASSES } from "../../inversify.types";
+import { TestConstants } from "../../TestConstants";
+import { By } from "selenium-webdriver";
+import 'reflect-metadata';
+
+
+@injectable()
+export class NewWorkspace {
+    private readonly driverHelper: DriverHelper;
+
+    private static readonly CHE_7_STACK_CSS: string = "div[data-stack-id='che7-preview']";
+    private static readonly SELECTED_CHE_7_STACK_CSS: string = ".stack-selector-item-selected[data-stack-id='che7-preview']"
+    private static readonly CREATE_AND_OPEN_BUTTON_XPATH: string = "(//che-button-save-flat[@che-button-title='Create & Open']/button)[1]"
+    private static readonly CREATE_AND_EDIT_BUTTON_CSS: string = "#dropdown-toggle button[name='dropdown-toggle']"
+    private static readonly ADD_OR_IMPORT_PROJECT_BUTTON_CSS: string = ".add-import-project-toggle-button";
+    private static readonly ADD_BUTTON_CSS: string = "button[aria-disabled='false'][name='addButton']";
+    private static readonly NAME_FIELD_CSS: string = "#workspace-name-input";
+
+
+    constructor(
+        @inject(CLASSES.DriverHelper) driverHelper: DriverHelper
+    ) {
+        this.driverHelper = driverHelper;
+    }
+
+    async selectCreateWorkspaceAndProceedEditing(timeout = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const createAndProceedEditingButtonLocator: By = By.xpath("//span[text()='Create & Proceed Editing']")
+
+        // open drop down list
+        await this.driverHelper.waitAndClick(By.css(NewWorkspace.CREATE_AND_EDIT_BUTTON_CSS), timeout)
+
+        // click on "Create & Proceed Editing" item in the drop down list
+        await this.driverHelper.waitAndClick(createAndProceedEditingButtonLocator, timeout)
+    }
+
+    async typeWorkspaceName(workspaceName: string, timeout = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const workspaceNameFieldLocator: By = By.css(NewWorkspace.NAME_FIELD_CSS)
+
+        await this.driverHelper.enterValue(workspaceNameFieldLocator, workspaceName, timeout)
+    }
+
+    async clickOnChe7Stack(timeout = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const che7StackLocator: By = By.css(NewWorkspace.CHE_7_STACK_CSS)
+
+        await this.driverHelper.waitAndClick(che7StackLocator)
+    }
+
+    async waitChe7StackSelected(timeout = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const che7SelectedStackLocator: By = By.css(NewWorkspace.SELECTED_CHE_7_STACK_CSS)
+
+        await this.driverHelper.waitVisibility(che7SelectedStackLocator, timeout)
+    }
+
+    async clickOnCreateAndOpenButton(timeout = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const ideFrameLocator: By = By.xpath("//ide-iframe[@id='ide-iframe-window' and @aria-hidden='false']")
```

Comment:
1. there should be just one element with given ID so the `aria-hidden` selector is probably not necessary
2. I recommend using CSS selectors as much as possible, in this case `By.css("ide-iframe#ide-iframe-window")`

Reply:
the "aria-hidden" selector is definitely important, it shows that IDE start loading